tests/phpunit/tests/rest-api/rest-plugins-controller.php

```diff
@@ -1036,6 +1041,20 @@ static function ( $reply, $package, $upgrader ) {
 			3
 		);
 
+		add_filter(
+			'plugins_api',
+			function ( $bypass, $action, $args ) {
+				// Only mock the plugin_information (link-manager) request.
+				if ( 'plugin_information' !== $action || 'link-manager' !== $args->slug ) {
+					return $bypass;
+				}
+				$http_response = file_get_contents( DIR_TESTDATA . '/plugins/link-manager.json' );
```

Comment:
In fact, rather than making this callback `static`, what about calling the `file_get_contents()` and `json_decode()` in `set_up_before_class()`, storing it to a property, and then this callback could just be:

```php
function ( $bypass, $action, $args ) {
	// Only mock the plugin_information (link-manager) request.
	if ( 'plugin_information' !== $action || 'link-manager' !== $args->slug ) {
		return $bypass;
	}
	return $this->decoded_http_response;
}
```

It would remove unnecessary repeat calls to `file_get_contents()` and `json_decode()`. Thoughts?

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/5998/commits/2565d25369243f55593a75389c5cb6eda34ba78b